index.js

```diff
@@ -1,55 +1,56 @@
-const express = require("express");
+const express = require('express');
 
-const PORT = 8080;
+const PORT = 8000;
 const app = express();
 
-var product = require('./src/data/products.json');
-var category = require('./src/data/categories.json');
+// Import dataService.js file.
+const dataService = require('./dataService');
 
-app.get('/', function (req, res) {
-    res.send(" ");
+// Create a route for app.
+app.get('/', function(req, res) {
+  res.send('');
 });
 
-app.get("/info", (req, res) => {
-    var serverInfo = require('./package.json');
-    res.json({serverName : serverInfo.name, serverVersion : serverInfo.version});
+// Display server information.
+app.get('/info', (req, res) => {
+  var serverInfo = require('./package.json');
+  res.json({serverName: serverInfo.name, serverVersion: serverInfo.version});
 });
 
-app.get("/products/all", (req, res) => {
-
-    // STEP 4. Combine category and product details, then send as JSON.
-
-    var combined = Object.assign({}, category, product);
-    res.json({allProducts : combined});
+// Send products and categories combined.
+app.get('/products/all', (req, res) => {
+  var combinedData = dataService.getCombinedProductMap();
+  let all = JSON.parse(JSON.stringify([...combinedData]));
+  res.json({all});
```

Comment:
You should just output `combinedData`. Format of the JSON output is

```json
{
  "P001": {
    "name": "...",
    "categoryId": "...",
    ...
  },
  "P002": {
    "name": "...",
    "categoryId": "...",
    ...
  }
}
```

Resubmit the screenshot when done.

Reply:
Fixed! The map could not be directly displayed, so I converted it into a JSON object and sent that for output.